File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
from typing import List

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i = 1  # Start from the second element
        while i < len(nums):
            if nums[i] == nums[i - 1]:
                del nums[i]  
            else:
                i += 1  
        return len(nums)

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int n = nums.length;
        if (n == 0) return 0;

        int k = 0;

        for (int i = 0; i < n - 1; i++) {
            if (nums[i] != nums[i + 1]) {
                nums[k] = nums[i];
                k++;
            }
        }

        // last element is always unique
        nums[k] = nums[n - 1];
        k++;

        return k;
    }
}
